lib/__tests__/cookieJar.spec.ts

```diff
@@ -1026,6 +1026,72 @@ describe('CookieJar', () => {
       ])
     })
   })
+
+  describe('Potentially Trustworthy Contexts', () => {
```

Comment:
This is missing a test for the `*.localhost` case. It wouldn't hurt to add cases for the `https` and `wss` protocols as well for completeness (even if those are already exercised in the existing test suite).